### DISCUSSION TITLE
Can I close popupmenu programmatically?

### DISCUSSION THREAD

**Discussion by reuse-ay** [maxGraph/maxGraph#405] [Q&A] [answered]
`MaxPopupMenu` has `hideMenu()`. Using this method, probably I can close menu. However, I don't know how to use this method because I can't find instance of derived class of `MaxPopupMenu`. In [Storybook sample](https://maxgraph.github.io/maxGraph/demo/?path=/story/misc-menustyle--default), menu gets available by including derived class of `MaxPopupMenu`  in argument of `MyCustomGraph`'s constructor as a part of `plugins`. `graph.plugins` has just plagin classes. This is what I understand. I have to find instance of derived class of `MaxPopupMenu`.
Or are there other way to close menu programmatically?

**Comment by reuse-ay** [comment, marked as answer]
Instance is in `graph.pluginsMap`
`(graph.pluginsMap[MyCustomPopupMenuHandler.pluginId] as MyCustomPopupMenuHandler).hideMenu();`

**Reply by tbouffard** [reply]
You also have a dedicated method in the Graph class that you can call to retrieve the plugin instance.
It is used in several stories and in the maxGraph code.
```ts
(graph.getPlugin('PopupMenuHandler') as MyCustomPopupMenuHandler).hideMenu();
```
